src/OpenApi/CarterOpenApi.cs

```diff
@@ -226,24 +200,38 @@ private static void CreateOpenApiResponseBody(OpenApiDocument document, KeyValue
                                     "application/json",
                                     new OpenApiMediaType
                                     {
-                                        //Example = respObj
                                         Schema = new OpenApiSchema { Reference = new OpenApiReference { Id = responseTypeName, Type = ReferenceType.Schema } }
-                                        //Schema = arrayType ? arrayschema : schema
                                     }
                                 }
                             }
                         };
 
                         if (!document.Components.Schemas.ContainsKey(responseTypeName))
                         {
+                            var example = CreateOpenApiExample(responseType, valueStatusCode.ResponseExample);
+
                             if (!arrayType)
                             {
+                                var propNames = responseType.GetProperties()
+                                    .Select(x => (Name: x.Name.ToLower(), Type: x.PropertyType.Name.ToLower()))
+                                    .ToList();
+
+                                var schema = new OpenApiSchema
+                                {
+                                    Type = "object",
+                                    Properties = propNames.ToDictionary(key => key.Name, value => new OpenApiSchema { Type = GetOpenApiTypeMapping(value.Type) }),
+                                    Example = example,
+                                };
+
                                 document.Components.Schemas.Add(responseTypeName, schema);
                             }
                             else
                             {
                                 document.Components.Schemas.Add(responseTypeName,
-                                    new OpenApiSchema { Type = "array", Items = new OpenApiSchema { Reference = new OpenApiReference { Id = singularTypeName, Type = ReferenceType.Schema } } });
+                                    new OpenApiSchema {
+                                            Type = "array",
+                                            Example = example,
```

Comment:
This needs to be an array of `example` or we decide we don't do examples for arrays of items as the arrays have a link to the singular item which has an example, see here https://github.com/CarterCommunity/Carter/blob/master/test/Carter.Samples.Tests/OpenAPITests.Should_return_Carter_approved_OpenAPI_json.approved.txt#L156 Thoughts?

Reply:
Any thoughts?